packages/wonder-blocks-core/src/hooks/__tests__/use-isomorphic-layout-effect.test.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+import {renderHook} from "@testing-library/react-hooks";
+import {renderToString} from "react-dom/server";
+
+import Server from "../../util/server";
+
+import {useIsomorphicLayoutEffect} from "../use-isomorphic-layout-effect";
+
+// We need to make useLayoutEffect spy-able.
+jest.mock("react", () => {
+    const ActualReact = jest.requireActual("react");
+    return {
+        ...ActualReact,
+        useLayoutEffect: jest.fn(),
+    };
+});
+
+describe("useIsomorphicLayoutEffect", () => {
+    describe("when client side", () => {
+        beforeEach(() => {
+            jest.spyOn(Server, "isServerSide").mockReturnValue(false);
+        });
+
+        it("should invoke React.useLayoutEffect with the given arguments", () => {
+            // Arrange
+            const useLayoutEffectSpy = jest.spyOn(React, "useLayoutEffect");
+            const effect = () => {};
+            const deps = ["foo", "bar"];
+
+            // Act
+            renderHook(() => useIsomorphicLayoutEffect(effect, deps));
+
+            // assert
+            expect(useLayoutEffectSpy).toHaveBeenCalledWith(effect, deps);
+        });
+    });
+
+    describe("when server side", () => {
+        beforeEach(() => {
+            jest.spyOn(Server, "isServerSide").mockReturnValue(true);
+        });
+
+        it("should not throw", () => {
+            // Arrange
+            const Component = () => {
+                useIsomorphicLayoutEffect(() => {}, []);
+                return null;
+            };
+
+            // Act
+            const underTest = () => renderToString(<Component />);
+
+            // Assert
+            expect(underTest).not.toThrowError();
+        });
```

Comment:
**question:** would `useLayoutEffect()` throw in the tests if `Server.isServerSide()` returns `true`?  Maybe we should check that `useLayoutEffect()` isn't being called in this scenario.

Reply:
It does give an error in those scenarios and I feel like adding a test for that is tautological in that we are adding this hook because it causes problems. So to then test that it causes problems seems odd to me.

Reply:
OK, so the production version of React throws an error. The development version logs an error instead. I can test for the latter, not the former. Updating accordingly.